pkg/payload/precondition/clusterversion/upgradeable.go

```diff
@@ -46,12 +47,18 @@ func (pf *Upgradeable) Run(ctx context.Context, releaseContext precondition.Rele
 
 	// if we are upgradeable==true we can always upgrade
```

Comment:
I always found this comment confusing, or at least overly simplistic, even before your changes. Can it be removed or enhanced.

Reply:
This comment is from #291, so orthogonal to my change.  Since this PR is a already a backport, I'd rather not pick it up here.  Feel free to file a separate master PR adjusting the wording.